cmd/realm/interfaces.go

```diff
@@ -235,46 +161,29 @@ func interfacesInstallF(command *cobra.Command, args []string) error {
 		return err
 	}
 
-	var requestBody struct {
-		Data map[string]interface{} `json:"data"`
-	}
-	requestBody.Data = interfaceBody
-
-	b := new(bytes.Buffer)
-	err = json.NewEncoder(b).Encode(requestBody)
-	if err != nil {
-		return err
-	}
-
-	req, err := http.NewRequest("POST", realmManagementUrl+"/v1/"+realm+"/interfaces", b)
-	if err != nil {
-		return err
-	}
-	req.Header.Add("Authorization", "Bearer "+realmManagementJwt)
-	req.Header.Add("Content-Type", "application/json")
-
-	resp, err := netClient.Do(req)
+	err = astarteAPIClient.RealmManagement.InstallInterface(realm, interfaceBody, realmManagementJwt)
 	if err != nil {
-		return err
+		fmt.Println(err)
+		os.Exit(1)
+		return nil
 	}
 
-	if resp.StatusCode == 201 {
-		fmt.Println("ok")
-	} else {
-		var errorBody struct {
-			Errors map[string]interface{} `json:"errors"`
-		}
+	fmt.Println("ok")
+	return nil
+}
 
-		err = json.NewDecoder(resp.Body).Decode(&errorBody)
-		if err != nil {
-			return err
-		}
+func interfacesDeleteF(command *cobra.Command, args []string) error {
+	interfaceName := args[0]
+	interfaceMajor := 0
 
-		errJson, _ := json.MarshalIndent(&errorBody, "", "  ")
-		fmt.Println(string(errJson))
+	err := astarteAPIClient.RealmManagement.DeleteInterface(realm, interfaceName, interfaceMajor, realmManagementJwt)
+	if err != nil {
+		fmt.Println(err)
 		os.Exit(1)
+		return nil
```

Comment:
Is this even reachable? You exit from the program the line above

Reply:
tbh that's a very good question and I had the same doubt

Reply:
From Exit docs:
The program terminates immediately; deferred functions are not run.

So yeah, they're useless.